src/mgxs_interface.cpp

```diff
@@ -0,0 +1,227 @@
+#include "mgxs_interface.h"
+
+namespace openmc {
+
+//==============================================================================
+// Mgxs data loading interface methods
+//==============================================================================
+
+void
+add_mgxs_c(hid_t file_id, const char* name, int energy_groups,
+     int delayed_groups, int n_temps, const double temps[],  double tolerance,
+     int max_order, bool legendre_to_tabular, int legendre_to_tabular_points,
+     int& method)
+{
+  //!! mgxs_data.F90 will be modified to just create the list of names
+  //!! in the order needed
+  // Convert temps to a vector for the from_hdf5 function
+  double_1dvec temperature;
+  temperature.assign(temps, temps + n_temps);
+
+  write_message("Loading " + std::string(name) + " data...", 6);
+
+  // Check to make sure cross section set exists in the library
+  hid_t xs_grp;
+  if (object_exists(file_id, name)) {
+    xs_grp = open_group(file_id, name);
```

Comment:
You need a corresponding call to `close_group` at the end of this function.

Reply:
- [x]